### ISSUE TITLE
Steam installscript.vdf checking other keys for EA App

### ISSUE THREAD

**Issue by Fractal147** [p0358/Fuck_off_EA_App#31]
### Checklist

- [X] I ensured I'm running the [latest version of the patch](https://github.com/p0358/Fuck_off_EA_App/releases), and that the issue is present in it if applicable
- [X] I have read the entirety of the [FAQ](https://github.com/p0358/Fuck_off_EA_App#faq) section in the readme, and my question was not answered there, or was not answered sufficiently
- [X] I have ran general troubleshooting steps (if applicable), such as restarting the PC/reinstalling Origin/whatever else is described in the [FAQ](https://github.com/p0358/Fuck_off_EA_App#faq), and they didn't help
- [X] I have checked [existing issues](https://github.com/p0358/Fuck_off_EA_App/issues?q=is%3Aissue) (including closed ones), and there wasn't an existing issue pertaining to my problem (with possible solutions in the comments)

### Your issue

Reinstalling today (proton 7 fwiw), Mass Effect Legendary repeatedly runs installscript.vdf, despite the "Pretend EA App is installed" checkbox.

Turns out Steam is checking a new key: and adding it in manually 'fixes' it.
```
"HasRunStringKey"               "HKEY_LOCAL_MACHINE\\SOFTWARE\\Electronic Arts\\EA Desktop\\InstallSuccessful"
"HasRunStringValue"             "true"
```

**Comment by p0358** [comment]
Hmm, thing is, the `HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Electronic Arts\EA Desktop\InstallSuccessful` in README is 64-bit registry, which corresponds to `HKEY_LOCAL_MACHINE\SOFTWARE\Electronic Arts\EA Desktop\InstallSuccessful` in 32-bit registry, and Steam was generally checking the 32-bit one. I wonder where's the disparity now then...

The installer should create that key if checkbox is ticked (in HKLM, meaning HKEY_LOCAL_MACHINE in 32-bit registry):
https://github.com/p0358/Fuck_off_EA_App/blob/3a612f86b1703c7cd687cd78e892ef6c9294f167/setup.iss#L46

Can you upload the full installScript.vdf of that game? I'll also need to check later if perhaps EA App would now create the keys in both places, if it doesn't, then the issue may be laying somewhere else...

**Comment by jkichler** [comment]
Can confirm same for me.  I checkbox is ticked for "Pretend EA App is installed".  HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Electronic Arts\EA Desktop\InstallSuccessful was created.   

HKEY_LOCAL_MACHINE\SOFTWARE\Electronic Arts\EA Desktop\InstallSuccessful was NOT created.

EA FC24 attempted to install ea app.

added HKEY_LOCAL_MACHINE\SOFTWARE\Electronic Arts\EA Desktop\InstallSuccessful manually

game launched from steam as expected with only origin install.

**Comment by Fractal147** [comment]
Certainly, though from what you've said it's probably not an issue with the FOEAA patch.
[installScript.vdf.txt](https://github.com/p0358/Fuck_off_EA_App/files/14089425/installScript.vdf.txt)

Perhaps some wine/proton/protontricks issue where somehow the the 64-bit keys are not getting duplicated to the 32-bit land? But dunno - it certainly used to work with no issue on 2023-08 ish. [No obvious changes to installscript or game since 2023-01](https://steamdb.info/app/1328670/patchnotes/). Hm!

Checked again right now with freshly validated game install, and new Proton Experimental prefix - same issue, installscript.vdf being run even after FOEAA patch.
